libsolutil/CommonData.h

```diff
@@ -141,6 +142,36 @@ inline std::multiset<T...>& operator-=(std::multiset<T...>& _a, C const& _b)
 namespace solidity::util
 {
 
+/// Functional map.
+/// Returns a container _oc applying @param _op to each element in @param _c.
+/// By default _oc is a vector.
+/// If another return type is desired, an empty contained of that type
+/// is given as @param _oc.
+template<class Container, class Callable, class OutputContainer =
+	std::vector<std::invoke_result_t<
+		Callable,
+		decltype(*std::begin(std::declval<Container>()))
+>>>
+auto fmap(Container const& _c, Callable&& _op, OutputContainer _oc = OutputContainer{})
```

Comment:
Why not call this `transform`? Because of the conflict with `std`?

Reply:
Rather because `transform` actually transforms something in place and this always returns a new object. I actually wanted to call it `map`, but most likely there would be problems with that

Reply:
`applyMap`? `mapElements`?